src/main/java/frc/robot/subsystems/DriveSub.java

```diff
@@ -0,0 +1,125 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix6.hardware.TalonFX;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+
+public class DriveSub extends SubsystemBase {
+
+    private TalonFX motor;
+    private SingleJointedArmSim armSim;
+    private Mechanism2d mech2d;
+    private MechanismLigament2d wheelLigament;
+    private PIDController pidController;
+
+    // Constructor
+
+    public DriveSub(int motorID) {
+
+        motor = new TalonFX(motorID);
+
+        if (RobotBase.isSimulation()) {
+            armSim = new SingleJointedArmSim(
+                    DCMotor.getFalcon500(1), // assuming Falcon500 motor
+                    1.0, // gear ratio of 1:1
+                    getMomentOfInertia(), // method for inertia
+                    getWheelRadius(), // method for wheel radius
+                    Double.NEGATIVE_INFINITY, // min angle
+                    Double.POSITIVE_INFINITY, // max angle
+                    false, // sim gravity false
+                    0); // disable gravity
+        }
+
+        // initialize mechanism2d and ligament
+        mech2d = new Mechanism2d(100, 100);
+        wheelLigament = new MechanismLigament2d("Wheel", 30, 0);
+
+        // intialize pid
+        pidController = new PIDController(0.03, 0, 0.11);
+        pidController.setTolerance(0.05);
```

Comment:
These values get the subsystem to a position accurately but not quickly. Try tuning more to get to the setpoint both quickly and accurately.

Reply:
Just tuned it. Might be too fast, though seems more accurate.